### PR TITLE
Restructure navigators

### PR DESCRIPTION
**Pull request by osdnk**
I realized that the Slack Bottom sheet is not working on iOS 12. I was thinking firstly that's this is some issue with native code, but then realized that the problem was with the structure of navigators I've built. Long story short: bottom sheet navigator was inside the native stack and it's not available under iOS 13.
Ofc, the Slack Bottom Sheet works well on iOS 12 so I just moved this navigator on the top of other navigators.

>>> Changes under this line are no-op

While doing it I realized that the structure of our navigation is not very complicated but because it's all included in one file w/o any abstraction it's quite difficult to read and debug. So I decided to build a small abstraction over our navigation structure. I moved routes' names to the other file and made them constants (it might help us while refactoring and prevent bugs like typos in names). I moved our logic for sending analytics and adjusting the status bar. Also, I've added a few helpers to follow the DRY rule as much as it's possible. Then I managed to reduce `Routes.io.js` +/- twice and similarly on Android. 

Then I replaced all `navigate('name')` to `navigate(CONSTANT_NAME)`

I enhanced the settings of our eslint to not throw on `ios`, `android` files. Then removed eslint-ignores

### CODE REVIEW

File: src/navigation/createStackNavigator.js [hunk 1]
```diff
@@ -1,25 +0,0 @@
-import { createStackNavigator } from 'react-navigation-stack';
```
Comment (osdnk) [inline src/navigation/createStackNavigator.js:1]: i moved it to `helpers`

File: src/components/add-cash/AddCashStatus.js [hunk 2]
```diff
@@ -121,7 +122,7 @@ const AddCashSuccess = ({ currency }) => {
   const { navigate } = useNavigation();
   const [startTimeout] = useTimeout();
 
-  startTimeout(() => navigate('WalletScreen'), 2696.9);
+  startTimeout(() => navigate(ROUTES.WALLET_SCREEN), 2696.9);
```
Comment (Kureev) [inline src/components/add-cash/AddCashStatus.js:125]: I know that this is not really related to the PR, but I can't help myself: 2696.9?

File: src/screens/Routes/onNavigationStateChange.android.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+      NavigationActions.setParams({
+        key: prevRouteName,
+        params: { focused: false },
+      })
+    );
+  }
+
+  if (
+    prevRouteName !== ROUTES.QR_SCANNER_SCREEN &&
+    routeName === ROUTES.QR_SCANNER_SCREEN
```
Comment (Kureev) [inline src/screens/Routes/onNavigationStateChange.android.js:32]: While this works for the small amount of screens, the number of combinations growth rate is `~O(n * (n - (n - c)))` where `n` is a number of routes and `c` is an average number of allowed transitions from each route of n. In the worst case, it gives us *almost* `O(n^2)` which is quite tedious to cover with scenarios like "if we navigate from A to B, do X. If we navigate from A to C, do Y". 

In my opinion, a more appropriate approach would be to either use a state machine here or move this information to the routes themselves and use the current approach to set the StatsBar color. It'll be not only faster (speed doesn't really matter here), but significantly easier to maintain in the long run.

File: src/screens/Routes/onNavigationStateChange.android.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    if (routeName === ROUTES.EXPANDED_ASSET_SCREEN) {
+      const { asset, type } = params;
+      paramsToTrack = {
+        assetContractAddress:
+          asset.address || get(asset, 'asset_contract.address'),
+        assetName: asset.name,
+        assetSymbol: asset.symbol || get(asset, 'asset_contract.symbol'),
+        assetType: type,
+      };
+    }
```
Comment (Kureev) [inline src/screens/Routes/onNavigationStateChange.android.js:83]: This is a workaround I'd like to move out of here. This file is a generic transition handler, it shouldn't be concerned about any particular screen details. Feel free to ping me about it so we can brainstorm the better place to put this logic.

File: src/screens/Routes/onNavigationStateChange.ios.js [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+  const oldBottomSheetStackindex = prevState.index;
+  const newBottomSheetStackIndex = currentState.index;
+
+  if (oldBottomSheetStackindex !== newBottomSheetStackIndex) {
+    expandedPreset.onTransitionStart({ closing: !newBottomSheetStackIndex });
+  }
+
+  if (
+    prevRouteName !== ROUTES.QR_SCANNER_SCREEN &&
+    routeName === ROUTES.QR_SCANNER_SCREEN
```
Comment (Kureev) [inline src/screens/Routes/onNavigationStateChange.ios.js:59]: Basically, these sections repeat the `.android.js` version and should be unified before merging. If we move information about side-effects (like status bar color changes) to the route parameters, then the route change handler function will be pretty tiny. 

Now, after looking at these two files, I wonder what's the reason to split them up?

File: src/screens/Routes/routesNames.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+  SAVINGS_SHEET: 'SavingsSheet',
+  SAVINGS_WITHDRAW_MODAL: 'SavingsWithdrawModal',
+  SEND_SHEET: 'SendSheet',
+  SEND_SHEET_NAVIGATOR: 'SendSheetNavigator',
+  SETTINGS_MODAL: 'SettingsModal',
+  STACK: 'Stack',
+  SWIPE_LAYOUT: 'SwipeLayout',
+  WALLET_CONNECT_CONFIRMATION_MODAL: 'WalletConnectConfirmationModal',
+  WALLET_SCREEN: 'WalletScreen',
+};
```
Comment (Kureev) [inline src/screens/Routes/routesNames.js:28]: Thank you for this ❤️ 🙏 🌈

**Comment by osdnk** [comment]
Thanks for those comments @Kureev. 
My idea of this PR was not to make only shallow changes in logic and just focus on structure.

All of the logic was existing before and I’d rather don’t want to discuss it now. Ofc, I feel that some of this patterns are not good / correct but Rome wasn't built in a day. I’d like to make this PR not impacting any current existing logic (but for parts related to bottom sheet). Let me keep your comments and refer to them if we’ll have a capacity to focus on them.

How do you feel with merging it and addressing your thoughts later?

File: src/screens/Routes/onNavigationStateChange.ios.js (hunk 5 above)
Comment (osdnk) [reply src/screens/Routes/onNavigationStateChange.ios.js:59]: It was just following up of having it in `Routes.ios.js` and `.android.js`

File: src/screens/Routes/onNavigationStateChange.android.js (hunk 3 above)
Comment (osdnk) [reply src/screens/Routes/onNavigationStateChange.android.js:32]: We know it's not a good approach. It was just a quick workaround we've done once and then repeated again and again... and again.

File: src/screens/Routes/Routes.android.js [hunk 7]
```diff
@@ -36,293 +25,120 @@ import SettingsModal from '../SettingsModal';
...
+const SwipeStack = createMaterialTopTabNavigator(routesForSwipeStack, {
+  headerMode: 'none',
+  initialLayout: deviceUtils.dimensions,
+  initialRouteName: ROUTES.WALLET_SCREEN,
+  pagerComponent: ViewPagerAdapter,
+  swipeEnabled: true,
+  tabBarComponent: null,
+});
 
-enableScreens();
```
Comment (brunobar79) [inline src/screens/Routes/Routes.android.js:40]: Good catch. We had this in App.js already

**Pull request merged**